File: scripts/R/example_script.R
# Load necessary libraries
library(ggplot2)
library(dplyr)

# Generate example data
set.seed(123)
data <- data.frame(
  category = rep(c("A", "B", "C"), each = 100),
  value = c(rnorm(100, mean = 5, sd = 2), 
            rnorm(100, mean = 7, sd = 2.5), 
            rnorm(100, mean = 6, sd = 1.5))
)

# Print the first few rows of the data
head(data)

# Summarize the data
summary_data <- data %>%
  group_by(category) %>%
  summarize(
    mean_value = mean(value),
    sd_value = sd(value)
  )

# Print the summary data
print(summary_data)

# Create a boxplot
ggplot(data, aes(x = category, y = value, fill = category)) +
  geom_boxplot() +
  labs(title = "Boxplot of Values by Category",
       x = "Category",
       y = "Value") +
  theme_minimal()

# Save the plot
ggsave("figures/boxplot.png")

File: scripts/python/example_script.py
# example_script.py

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Generate example data
np.random.seed(123)
data = pd.DataFrame(
    {
        "category": np.repeat(["A", "B", "C"], 100),
        "value": np.concatenate(
            [
                np.random.normal(5, 2, 100),
                np.random.normal(7, 2.5, 100),
                np.random.normal(6, 1.5, 100),
            ]
        ),
    }
)

# Print the first few rows of the data
print(data.head())

# Summarize the data
summary_data = (
    data.groupby("category")
    .agg(mean_value=("value", "mean"), sd_value=("value", "std"))
    .reset_index()
)

# Print the summary data
print(summary_data)

# Create a boxplot
plt.figure(figsize=(10, 6))
data.boxplot(column="value", by="category", grid=False)
plt.title("Boxplot of Values by Category")
plt.suptitle("")
plt.xlabel("Category")
plt.ylabel("Value")

# Save the plot
plt.savefig("boxplot.png")
plt.show()
